test/sample-test/sample_test_launcher.py

```diff
@@ -97,29 +94,72 @@ def _compile_sample(self):
     os.chdir(self._work_dir)
     print('Run the sample tests...')
 
+    # Looking for the entry point of the test.
+    list_of_files = os.listdir('.')
+    for file in list_of_files:
```

Comment:
Why listing all files? isn't it  the convention that sample python name is exactly the test_name?

Reply:
It is, but I want to catch the error when test_name.py and test_name.ipynb exist at the same time. If that's indeed the case, the correct behavior is not defined.

Reply:
What if the test name is actually the test file name? Then there would be no ambiguity.

Reply:
> What if the test name is actually the test file name? Then there would be no ambiguity.

That 's something doable. However the current convention is that the dir name should also be consistent with the test name, which should not contain an extension. Also, calling a test something like a-simple-kfp-test.ipynb is a bit of weird.

Maybe this can be solved in a more elegant way, for example, designate the entrypoint in the config.

Reply:
>However the current convention is that the dir name should also be consistent with the test name

What if we want to put several notebooks into one directory? There are some real use-cases for this. (per-component notebooks + pipeline notebook; saving/sharing component + loading/consuming component).

> Also, calling a test something like a-simple-kfp-test.ipynb is a bit of weird.

I think tests could be named any way (spaces, capital letters, etc)

Reply:
> What if we want to put several notebooks into one directory?
Current convention is that the entrypoint is automatically selected as the one with the same name as the dir, which means only one notebook will be run. Indeed this is a case that's not covered by current sample test infra. Need some further thoughts on how to implement this scenario.

Reply:
Agreed. The current convention is the test name being the same as the directory name. In case of multiple test files or multi-layer hierarchy, we will need to improve the test infra later.